### ISSUE TITLE
function_code and len types

### ISSUE THREAD

**Issue by freitagbr** [freitagbr/cr-multihash#6]
The `function_code` and `len` attributes of the `Multihash` class are `Int32`. Many of the other implementations, like [Go](https://github.com/multiformats/go-multihash/blob/master/multihash.go#L74), have these attributes as uint64. Our attributes should be changed to `UInt64`, to better match the other implementations.

**Comment by nfischer** [comment]
Shouldn't these be Unsigned VarInts? That's [what the spec says](http://multiformats.io/multihash/#the-multihash-format)

**Comment by freitagbr** [comment]
Yeah, I guess they should be. I added the `uvarint` dependency so we can just use that.
